pkg/pdatatest/pmetrictest/options.go

```diff
@@ -179,6 +180,90 @@ func IgnoreMetricAttributeValue(attributeName string, metricNames ...string) Com
 	})
 }
 
+// IgnoreMetricAttributeValue is a CompareMetricsOption that clears value of the metric attribute.
+func IgnoreDatapointAttributesOrder() CompareMetricsOption {
+	return compareMetricsOptionFunc(func(expected, actual pmetric.Metrics) {
+		_ = orderDatapointAttributes(expected)
+		_ = orderDatapointAttributes(actual)
+	})
+}
+
+func orderDatapointAttributes(metrics pmetric.Metrics) error {
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		ilms := rms.At(i).ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			msl := ilms.At(j).Metrics()
+			for g := 0; g < msl.Len(); g++ {
+				msl.At(g)
+				switch msl.At(g).Type() {
+				case pmetric.MetricTypeGauge:
+					for k := 0; k < msl.At(g).Gauge().DataPoints().Len(); k++ {
+						rawOrdered := orderMapByKey(msl.At(g).Gauge().DataPoints().At(k).Attributes().AsRaw())
+						err := msl.At(g).Gauge().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
```

Comment:
I may be mistaken but I think we _should_ be able to assume that there aren't errors here because we're effectively just recreating the map from itself. Given that, and because this is test code, I think we should panic instead of returning an error.

Reply:
true, adapted!